app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild

```diff
@@ -0,0 +1,88 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_5,3_6} )
+
+DISTUTILS_OPTIONAL=1
+
+inherit distutils-r1 qmake-utils
+
+DESCRIPTION="Hex editor library, Qt application written in C++ with Python bindings"
+HOMEPAGE="https://github.com/lancos/qhexedit2/"
+SRC_URI="https://github.com/lancos/${PN}/archive/5f3ca79dbe8c765b685636a1751d242a6bfabb7d.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc +gui python"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.8.4-fixtest.patch"
+	"${FILESDIR}/${PN}-0.8.4-setup.py.patch"
+)
+
+RDEPEND="
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtwidgets:5
+	python? (
+		dev-python/PyQt5[gui,widgets,${PYTHON_USEDEP}]
+		dev-python/sip[${PYTHON_USEDEP}]
+		${PYTHON_DEPS}
+		)
+"
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/${PN}-5f3ca79dbe8c765b685636a1751d242a6bfabb7d"
```

Comment:
instead of this, consider using `vcs-snapshot.eclass`

Reply:
@SoapGentoo, please don't encourage use of deprecated eclasses. Since the hash needs to be known anyway, it's better to use it explicitly.

@ktrace, just move it to a common variable `EGIT_COMMIT` and reuse it both for SRC_URI and S.

Reply:
fixed

Reply:
@ktrace but please reuse `${EGIT_COMMIT}` here.

Reply:
Also this.